### PR TITLE
README for metrics spec

### PR DESCRIPTION
**Pull request by reyang**
## Changes

1. Have a general README doc to guide readers through the metrics specs.
2. Provide context - e.g. design goal.

Eventually we want to move the metrics terms/concepts here, so the API/SDK spec can focus on the concrete classes/methods/requirements.

Related issues #

Related [oteps](https://github.com/open-telemetry/oteps):
* https://github.com/open-telemetry/oteps/blob/main/text/metrics/0146-metrics-prototype-scenarios.md

### CODE REVIEW

File: specification/metrics/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  ability to achieve the same functionality as their native clients.
+
+### Concepts
+
+#### API
+
+The **OpenTelemetry Metrics API** ("the API" hereafter) serves two purposes:
+
+* Capturing raw measurements efficiently and simultaneously.
+* Allowing [SDK](#sdk) injection at runtime.
```
Comment (jsuereth) [inline specification/metrics/README.md:71]: There may be a better way to phrase this.

E.g. Decoupling "instrumentation" from "instrumentation pipeline".

SDK Injection is just one technique to this.

File: specification/metrics/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+If there is no [SDK](#sdk) injected, all the data reported via the API will be
+dropped on the floor. Please refer to the overall [OpenTelemetry
```
Comment (jsuereth) [inline specification/metrics/README.md:74]: nit;  'all data will be efficiently ignored'.  The closer to a  "no-op" we make this, the better for framework adoption.

Comment (jmacd) [reply specification/metrics/README.md:74]: The term "minimal implementation" is used here:

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/library-guidelines.md#api-and-minimal-implementation

File: specification/metrics/README.md (hunk 1 above)
Comment (jmacd) [reply specification/metrics/README.md:71]: The Library guidelines make a very clear statement on this topic, https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/library-guidelines.md

**Review by jmacd — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

File: specification/metrics/README.md (hunk 1 above)
Comment (reyang) [reply specification/metrics/README.md]: I've updated the wording and also added link to the library guidelines as @jmacd mentioned.
@jsuereth I reworded it to "decoupling" but also called out "allowing the SDK to be specified/included in the application" explicitly.

File: specification/metrics/README.md (hunk 2 above)
Comment (reyang) [reply specification/metrics/README.md]: Updated.

**Comment by reyang** [comment]
PTAL.

File: specification/metrics/README.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+#### API
+
+The **OpenTelemetry Metrics API** ("the API" hereafter) serves two purposes:
+
+* Capturing raw measurements efficiently and simultaneously.
+* Decoupling the instrumentation from the [SDK](#sdk), allowing the SDK to be
+  specified/included in the application.
+
+When no [SDK](#sdk) is explicitly included in the application, no telemetry data
```
Comment (cijothomas) [inline specification/metrics/README.md:75]:
```suggestion
When no [SDK](#sdk) is explicitly included/enabled in the application, no telemetry data
```

File: specification/metrics/README.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  traces can be correlated via exemplars, and metrics dimensions can be enriched
+  via [Baggage](../baggage/api.md) and [Context](../context/context.md).
+  Additionally, [Resource](../resource/sdk.md) can be applied to
+  [logs](../overview.md#log-signal)/[metrics](../overview.md#metric-signal)/[traces](../overview.md#tracing-signal)
+  in a consistent way.
+
+* **Providing a path for [OpenCensus](https://opencensus.io/) customers to
+  migrate to OpenTelemetry**. This was the original goal of OpenTelemetry -
+  converging OpenCensus and OpenTracing. We will focus on providing the
+  semantics and capability, instead of doing a 1-1 mapping of the APIs.
```
Comment (ThomsonTan) [inline specification/metrics/README.md:55]: "Semantics and capabilities"? For "providing the semantics and capability", does it mean the instrumentation code may need change?

Comment (ThomsonTan) [reply specification/metrics/README.md:55]: Please ignore my comment on "Semantics and capabilities", I incorrectly assumed semantics is in plural form.

**Pull request merged**